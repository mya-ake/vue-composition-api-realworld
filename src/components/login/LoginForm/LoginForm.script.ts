import { createComponent, reactive } from '@vue/composition-api';
import { BaseInput, BaseButton } from '@/components/base';
import { LoginFormType } from './LoginForm.type';

export default createComponent({
  components: {
    BaseInput,
    BaseButton,
  },
  props: {
    disabled: { default: false },
  },
  setup(_, { emit }) {
    const values = reactive<LoginFormType.Values>({ email: '', password: '' });
    const onSubmit = () => {
      emit('submit', { ...values });
    };
    return {
      values,
      onSubmit,
    };
  },
});
