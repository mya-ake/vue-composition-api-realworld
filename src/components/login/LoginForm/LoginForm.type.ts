export namespace LoginFormType {
  export interface Props {
    disabled: boolean;
  }

  export interface Values {
    email: string;
    password: string;
  }
}
