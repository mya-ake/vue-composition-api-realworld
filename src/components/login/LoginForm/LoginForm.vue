<template>
  <form @submit.prevent="onSubmit">
    <fieldset class="form-group" :disabled="disabled">
      <BaseInput
        v-model="values.email"
        type="email"
        class="form-control-lg"
        placeholder="Email"
      />
    </fieldset>
    <fieldset class="form-group" :disabled="disabled">
      <BaseInput
        v-model="values.password"
        type="password"
        class="form-control-lg"
        placeholder="Password"
      />
    </fieldset>
    <BaseButton
      :disabled="disabled"
      type="submit"
      class="btn-lg btn-primary pull-xs-right"
    >
      Sign in
    </BaseButton>
  </form>
</template>

<script lang="ts" src="./LoginForm.script.ts"></script>
