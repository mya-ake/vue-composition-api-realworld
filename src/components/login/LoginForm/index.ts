export { default as LoginForm } from './LoginForm.vue';
export * from './LoginForm.type';
