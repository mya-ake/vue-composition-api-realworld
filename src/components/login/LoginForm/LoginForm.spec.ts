import { shallowMount, Wrapper } from '@vue/test-utils';
import LoginForm from './LoginForm.vue';

describe('components/login/LoginForm', () => {
  describe('mountable', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LoginForm, {});
    });

    it('is vue instance', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
