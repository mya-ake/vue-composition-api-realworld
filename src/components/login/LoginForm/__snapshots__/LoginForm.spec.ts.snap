// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/login/LoginForm mountable snapshot 1`] = `
<form>
  <fieldset
    class="form-group"
  >
    <baseinput-stub
      class="form-control-lg"
      placeholder="Email"
      type="email"
      value=""
    />
  </fieldset>
   
  <fieldset
    class="form-group"
  >
    <baseinput-stub
      class="form-control-lg"
      placeholder="Password"
      type="password"
      value=""
    />
  </fieldset>
   
  <basebutton-stub
    class="btn-lg btn-primary pull-xs-right"
    type="submit"
  >
    
    Sign in
  
  </basebutton-stub>
</form>
`;
